### PR TITLE
ARF 80244 Update POARequestsTable to align with ARP engine mock data structure

### PR DESCRIPTION
**Pull request by williamphelps13**
## Summary

- Now that the ARP engine returns mock data, we need to align with that _data structure_ in the [ARP POARequestsTable](https://github.com/department-of-veterans-affairs/vets-website/blob/35ccb9675a553d8dda75b3d083ae662b3179662f/src/applications/accredited-representative-portal/components/POARequestsTable/POARequestsTable.jsx).
- [This PR](https://github.com/department-of-veterans-affairs/vets-website/pull/29034/files) updates our components to use that _data structure_.
- Changes:
  -  Rename POARequests.jsx to POARequestsPage.jsx. This should be our new standard to include the word page for each full page component.
  -  Add POARequestsTableFetcher.jsx which aligns with the `cell` pattern found in the RedwoodJS framework in which there is a component that contains the API call that then returns loading, error, empty, or success components based on the outcome of the API call.
  -  Create generic Error Message component that we will likely need to update or make dynamic in the future
  - Create custom hook usePOARequests.ts that contains useEffect and does the API calling itself NOTE: in the finally the poaRequests are set to the mockPOARequests data and the error is set to null. There is a To Do that will be completed after Auth is implemented.
  - Updated poaRequests.js functions to uses correct endpoint syntax (Gabe and I discovered this update when pairing on Sign In)
  -  Remove try/catch from poaRequests.js functions since it is necessary to have this in the component itself (or the custom hook the component uses) in order to save the loading, error, and data in state
  - Update poaRequests.unit.spec.js to use the standard constant syntax and to handle the removal of the try/catch from poaRequests.js
  - Update POARequestsTable.jsx and POARequestsWidgetTable.jsx after updated mockPOARequests.js to align with data structure in ARP engine. Also took some steps to align the POA Requests Table with the design:
image

## Related issue(s)

- [#80244 Use the LH API endpoint data structure in the POARequestsTable](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/80244)

#### Mobile - Before
Screenshot 2024-05-06 at 9 40 55 AM

#### Mobile - After
Screenshot 2024-05-06 at 9 41 16 AM

#### Desktop - Before
Screenshot 2024-05-06 at 9 41 34 AM

#### Desktop - After
Screenshot 2024-05-06 at 9 41 50 AM

## Testing done
- Power of attorney in heading still links to POA Requests
- Sidebar POA Requests still links to POA Requests
- The user with id 7482299145 links to representative/poa-requests/7482299145
- Format of cells matches design

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
Except new error: http://localhost:3000/accredited_representative_portal/v0/power_of_attorney_requests?poa_codes=A1Q net::ERR_CONNECTION_REFUSED which is expected because of Auth (will be fixed with Sign In issue)

### CODE REVIEW

**Review by gabezurita — COMMENTED** [review]
I will review this later today, but to clarify, the intent here is more to update the POARequestsTable to align with the ARP Engine mock data (we'll update both the engine and frontend to align with the LH updates in [a follow-on issue](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/81653))

Also, the linked ZH issue needs to be corrected! The below is the right one.
https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/80244

**Review by cohnjesse — DISMISSED** [review]
Just some small non-blocking stuff, loaded this up locally though and it looks great!

**Review by cohnjesse — DISMISSED** [review]
Reapproving

File: src/applications/accredited-representative-portal/components/POARequestsTable/POARequestsTable.jsx [hunk 1]
```diff
@@ -1,70 +1,111 @@
...
+export const createLimitationsCell = (healthInfoAuth, changeAddressAuth) => {
+  let text = null;
+  // If do not authorize sharing health info or authorize change of address then we label it as a limitation of consent
+  if (healthInfoAuth === 'N' && changeAddressAuth === 'N') {
+    text = 'Health, Address';
+  } else if (healthInfoAuth === 'N') {
+    text = 'Health';
+  } else if (changeAddressAuth === 'N') {
+    text = 'Address';
+  }
```
Comment (gabezurita) [inline src/applications/accredited-representative-portal/components/POARequestsTable/POARequestsTable.jsx:22]: Could simplify this logic (please test the below):
```
export const createLimitationsCell = (healthInfoAuth, changeAddressAuth) => {
  const limitations = [];
  if (healthInfoAuth === 'N') limitations.push('Health');
  if (changeAddressAuth === 'N') limitations.push('Address');

  return limitations.length > 0 ? (
    <span className="limitations-row">
      <va-icon 
        class="limitations-row__warning-icon"
        icon="warning"
        size={3}
        srtext="warning"
      />
      {limitations.join(', ')}
    </span>  
  ) : 'None';
};
```

**Review by gabezurita — COMMENTED** [review]
Found a thing you could try to simplify:

File: src/applications/accredited-representative-portal/sass/accredited-representative-portal.scss [hunk 2]
```diff
@@ -37,3 +37,19 @@
...
+  &__warning-icon {
+    color: var(--vads-color-action-focus-on-light);
+  }
+}
+
+.relationship-row {
+  display: block;
+  font-size: 1.4rem;
+  line-height: 1;
+}
```
Comment (gabezurita) [inline src/applications/accredited-representative-portal/sass/accredited-representative-portal.scss:55]: Missing newline!

File: src/applications/accredited-representative-portal/components/common/ErrorMessage.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    <div>
+      <p className="vads-u-margin-y--0">
+        Please refresh this page or check back later. You can also check the
+        system status on the VA.gov homepage.
+      </p>
+    </div>
+  </va-alert>
+);
+
+export default ErrorMessage;
```
Comment (gabezurita) [inline src/applications/accredited-representative-portal/components/common/ErrorMessage.jsx:15]: Nice and dry!

File: src/applications/accredited-representative-portal/containers/POARequestsPage.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+import POARequestsTableFetcher from '../components/POARequestsTableFetcher/POARequestsTableFetcher';
+import usePOARequests from '../hooks/usePOARequests';
+
+const POARequestsPage = () => {
+  return (
+    <>
+      <h1 data-testid="poa-requests-heading">Power of attorney requests</h1>
+      <POARequestsTableFetcher usePOARequests={usePOARequests} />
```
Comment (gabezurita) [inline src/applications/accredited-representative-portal/containers/POARequestsPage.jsx:10]: Nice, this is a rather OO approach!

File: src/applications/accredited-representative-portal/tests/actions/poaRequests.unit.spec.js [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
 import {
   acceptPOARequest,
   declinePOARequest,
   getPOARequestsByCodes,
 } from '../../actions/poaRequests';
-import environment from '~/platform/utilities/environment';
+import mockResponseData from '../../mocks/mockPOARequestsResponse.json';
 
-const prefix = `${environment.API_URL}/v0/accredited_representative_portal`;
+const API_PREFIX = `${environment.API_URL}/accredited_representative_portal/v0`;
```
Comment (gabezurita) [inline src/applications/accredited-representative-portal/tests/actions/poaRequests.unit.spec.js:13]: nice clearer renaming

File: src/applications/accredited-representative-portal/tests/components/POARequestsTableFetcher.unit.spec.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      poaRequests: mockPOARequests,
+    });
+    const { getByTestId } = render(
+      <MemoryRouter>
+        <POARequestsTableFetcher usePOARequests={usePOARequestsMock} />
+      </MemoryRouter>,
+    );
+    expect(getByTestId('poa-requests-table')).to.exist;
+  });
+});
```
Comment (gabezurita) [inline src/applications/accredited-representative-portal/tests/components/POARequestsTableFetcher.unit.spec.js:61]: Well tested!

**Review by gabezurita — APPROVED** [review]
LGTM!

Super clean, OO approach!

**Pull request merged**